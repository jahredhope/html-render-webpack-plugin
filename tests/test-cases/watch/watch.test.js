const MemoryFS = require("memory-fs");
const webpack = require("webpack");
const path = require("path");

const config = require("./webpack.config");
const HtmlRenderPlugin = require("../../../src");
const getDirContentsSync = require("../../utils/getDirContentsSync");

describe("Watch", () => {
  const renderDirectory = path.join(process.cwd(), "dist", "render");

  it("should render a HTML on initial watch build", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    compiler.apply(
      new HtmlRenderPlugin({
        routes: [{ route: "/a/", extra: "a" }, { route: "/b/", extra: "b" }],
        renderDirectory
      })
    );

    const watching = compiler.watch({}, error => {
      watching.close();
      expect(error).toBe(null);
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();
      done();
    });
  });
});
