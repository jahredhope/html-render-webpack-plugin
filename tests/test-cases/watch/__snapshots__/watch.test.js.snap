// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watch should render a HTML on initial watch build 1`] = `
Object {
  "a": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  common: {
  \\"value\\": \\"a\\"
}
  <code>[
    {
      \\"route\\": \\"/a/\\",
      \\"extra\\": \\"a\\"
    }
  ]</code>
</body>
</html>",
  },
  "b": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  common: {
  \\"value\\": \\"a\\"
}
  <code>[
    {
      \\"route\\": \\"/b/\\",
      \\"extra\\": \\"b\\"
    }
  ]</code>
</body>
</html>",
  },
}
`;
